save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -168,69 +178,94 @@ class DownloadProjectController(private val configProperties: ConfigProperties)
     private fun sendToBackendAndOrchestrator(
         project: Project,
         propertiesRelativePath: String,
-        resourcesRootPath: String,
+        resourcesRootRelativePath: String,
         executionType: ExecutionType,
-        testSuitesDto: List<TestSuiteDto>?
+        testSuiteDtos: List<TestSuiteDto>?
     ) {
-        testSuitesDto?.let {
-            require(executionType == ExecutionType.STANDARD)
-        } ?: require(executionType == ExecutionType.GIT)
+        testSuiteDtos?.let {
+            require(executionType == ExecutionType.STANDARD) { "Test suites shouldn't be provided unless ExecutionType is STANDARD (actual: $executionType)" }
+        } ?: require(executionType == ExecutionType.GIT) { "Test suites are not provided, but should for executionType=$executionType" }
```

Comment:
Do we really need this `require`, if we have verification in frontend to pass only one type of project?

Reply:
I believe, additional validation won't hurt, since we don't have good integration tests or tests on frontend yet